agent/proxycfg/upstreams.go

```diff
@@ -301,14 +301,13 @@ func (s *handlerUpstreams) resetWatchesFromChain(
 		delete(snap.WatchedUpstreams[uid], targetID)
 		delete(snap.WatchedUpstreamEndpoints[uid], targetID)
 		cancelFn()
-
-		targetUID := NewUpstreamIDFromTargetID(targetID)
-		if targetUID.Peer != "" {
-			snap.PeerUpstreamEndpoints.CancelWatch(targetUID)
-			snap.UpstreamPeerTrustBundles.CancelWatch(targetUID.Peer)
-		}
 	}
-
+	reconcilePeeringWatches(snap.DiscoveryChain,
```

Comment:
I think this would be clearer if it were moved out of `resetWatchesFromChain` to be called serially afterwards, since it isn't chain-specific.

Reply:
There is a dependency because we init the watch inside this function and that needs to be done after calling the reconciliation.